### PR TITLE
Restrict VLM padding workaround to transformers 5.3.0

### LINKED ISSUE #5502: Warning log: Kwargs passed to `processor.__call__` have to be in `processor_kwargs` dict, not in `**kwargs`
While training we get MANY warning logs: https://github.com/huggingface/trl/actions/runs/24199149472/job/70637611379
> Kwargs passed to `processor.__call__` have to be in `processor_kwargs` dict, not in `**kwargs`

After investigation, it seems the issue appeared with `transformers` v5.4.0 after the merge of this PR:
- https://github.com/huggingface/transformers/pull/44881

### PR DESCRIPTION
**Pull request by albertvillanova**
Restrict VLM padding workaround to transformers 5.3.0.

This PR updates the prompt tokenization logic in several trainer classes to handle a specific bug in the `transformers` library more robustly. The main change is to apply a padding workaround only for affected `transformers` versions (5.3.x), and to simplify the logic for handling padded and unpadded input IDs. This avoids unnecessary padding for unaffected versions and ensures compatibility with future releases.

Additionally, this PR fixes:
- #5502
- this issue arises when passing `padding=True`, but only with transformers >= 5.4.0 

Fix #5502.

### Motivation

- The bug (`Qwen2_5_VLProcessor.apply_chat_template` crashing on batched unpadded input) was introduced in transformers 5.3.0:
  - The original comment also incorrectly attributed the bug to transformers 5.2.0.
  - https://github.com/huggingface/transformers/issues/44514

- The bug was fixed in 5.4.0
  - huggingface/transformers#44563

- In `trl`, it was fixed with `padding=True` workaround in `_tokenize_prompts` (GRPO, RLOO, DPPO trainers) that was applied unconditionally to all transformers versions
  - #5239

### Solution

- Gate the workaround with `Version("5.3.0") <= Version(transformers.__version__) < Version("5.4.0")`
- Pass `padding=True` only when the workaround is active; omit the argument entirely otherwise (see #5502)
- Skip the unpadding step when the workaround is not active, using `tokenized["input_ids"]` directly
- Correct the comment to reference the right version and link the fix PR

### Changes

**Bug workaround and version handling:**

* Added a conditional check for the `transformers` library version (>=5.3.0 and <5.4.0) to determine if the padding workaround should be applied, instead of always applying padding. (GRPO, RLOO, and experimental DPPO)
* Updated comments to clarify that the bug is present in `transformers` 5.3.0 and fixed in 5.4.0, with references to the relevant GitHub issues and PRs. 

**Prompt ID extraction logic:**

* Only unpads `input_ids` using the `attention_mask` when the workaround is needed; otherwise, uses the tokenized `input_ids` directly, simplifying the code path for unaffected versions.

---

> [!NOTE]
> **Medium Risk**
> Touches prompt tokenization in multiple trainers; version-gated padding/unpadding could change input IDs for some `transformers` versions and affect multimodal generation edge cases.
> 
> **Overview**
> Restricts the VLM `apply_chat_template` padding workaround to `transformers` versions `>=5.3.0` and `<5.4.0`, instead of always forcing `padding=True`.
> 
> When the workaround is inactive, the trainers now skip the unpadding step and use `tokenized["input_ids"]` directly; comments were updated to reference the correct upstream bug/fix (transformers#44514/#44563).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cf3dbe2adccdb5a7b60d7eedf77d82271fa6536b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/dppo/dppo_trainer.py [hunk 1]
```diff
@@ -246,23 +246,28 @@ def _tokenize_prompts(self, prompts: list):
                 images.append(prompt_images if prompt_images else None)
             images = images if has_images else None
 
-            # We pass padding=True to work around a bug introduced in transformers 5.2.0 in some processors
-            # (e.g. Qwen2.5-VL) that crash on batched unpadded input. We then unpad input_ids using attention_mask.
-            # See: https://github.com/huggingface/transformers/issues/44514
+            # Workaround for a bug in transformers 5.3.0 where some processors (e.g. Qwen2.5-VL) crash on
+            # batched unpadded input (transformers#44514).
+            # Fixed in transformers 5.4.0 (transformers#44563).
+            needs_padding_workaround = Version("5.3.0") <= Version(transformers.__version__) < Version("5.4.0")
```
Comment (qgallouedec) [inline trl/experimental/dppo/dppo_trainer.py:252]: So we were wrong, it was introduced in 5.3 not 5.2?

**Review by qgallouedec — APPROVED** [review]

File: trl/experimental/dppo/dppo_trainer.py (hunk 1 above)
Comment (albertvillanova) [reply trl/experimental/dppo/dppo_trainer.py:252]: Yes, I tested locally.
> The original comment also incorrectly attributed the bug to transformers 5.2.0
